cheroot/workers/threadpool.py

```diff
@@ -223,6 +234,7 @@ def shrink(self, amount):
         # to remove. As each request is processed by a worker, that worker
         # will terminate and be culled from the list.
         for n in range(n_to_remove):
```

Comment:
Even better: you can move it out of the loop:
```suggestion
        self._pending_shutdowns.extend((_SHUTDOWNREQUEST, ) * n_to_remove)
        for n in range(n_to_remove):
```

Reply:
If it would mean that the loop was no longer necessary, then I would. But it doesn't, so I think it's better to keep the loop, as it reads easier.